File: core/spoofax.core/src/main/java/mb/spoofax/core/platform/PieLogger.java
package mb.spoofax.core.platform;

import mb.pie.api.Logger;
import org.checkerframework.checker.nullness.qual.Nullable;

public class PieLogger implements Logger {
    private final mb.log.api.Logger logger;

    public PieLogger(mb.log.api.Logger logger) {
        this.logger = logger;
    }

    @Override
    public void error(String message, @Nullable Throwable throwable) {
        logger.error(message, throwable);
    }

    @Override
    public void warn(String message, @Nullable Throwable throwable) {
        logger.warn(message, throwable);
    }

    @Override
    public void info(String message) {
        logger.info(message);
    }

    @Override
    public void debug(String message) {
        logger.debug(message);
    }

    @Override
    public void trace(String message) {
        logger.trace(message);
    }
}

File: core/spoofax.compiler.gradle/src/main/kotlin/mb/spoofax/compiler/gradle/PieLogger.kt
package mb.spoofax.compiler.gradle

import org.gradle.api.logging.Logger

class PieLogger(private val logger: Logger) : mb.pie.api.Logger {
  override fun error(message: String, throwable: Throwable?) {
    logger.error(message, throwable)
  }

  override fun warn(message: String, throwable: Throwable?) {
    logger.warn(message, throwable)
  }

  override fun info(message: String) {
    logger.info(message)
  }

  override fun debug(message: String) {
    logger.debug(message)
  }

  override fun trace(message: String) {
    logger.trace(message)
  }
}
